snuba/migrations/snuba_migrations/metrics/0025_metrics_counters_aggregate_v2.py

```diff
@@ -0,0 +1,100 @@
+from typing import Sequence
+
+from snuba.clickhouse.columns import Array, Column, UInt
+from snuba.clusters.storage_sets import StorageSetKey
+from snuba.datasets.storages.tags_hash_map import INT_TAGS_HASH_MAP_COLUMN
+from snuba.migrations import migration, operations, table_engines
+from snuba.migrations.columns import MigrationModifiers as Modifiers
+from snuba.migrations.snuba_migrations.metrics.templates import COMMON_AGGR_COLUMNS
+from snuba.utils.schemas import AggregateFunction, Float, String
+
+
+class Migration(migration.ClickhouseNodeMigration):
+    blocking = False
+    table_name = "metrics_counters_v2_local"
+    dist_table_name = "metrics_counters_v2_dist"
+    granularity = "2048"
+    aggregated_cols: Sequence[Column[Modifiers]] = [
+        # This is a common column for aggregate tables going forward
+        # but doesn't exist in the old templates so we include it here
+        Column("use_case_id", String(Modifiers(low_cardinality=True))),
+        *COMMON_AGGR_COLUMNS,
+        Column("value", AggregateFunction("sum", [Float(64)])),
+    ]
+
+    def forwards_local(self) -> Sequence[operations.SqlOperation]:
+        return [
+            operations.CreateTable(
+                storage_set=StorageSetKey.METRICS,
+                table_name=self.table_name,
+                columns=self.aggregated_cols,
+                engine=table_engines.AggregatingMergeTree(
+                    storage_set=StorageSetKey.METRICS,
+                    order_by="(org_id, project_id, metric_id, granularity, timestamp, use_case_id, tags.key, tags.value)",
```

Comment:
retention is still missing.
I thought `use_case_id` would be the first in the key. 
If we start using use_case_id we will always filter by that, so we should put it both in the order by and the primary key. 

The fields we have in the order by but not in the primary key are not part of the index, so we do not get any advantage there

Reply:
I'm fine with use_case_id being part of the `primary key` and `order by`, but was this comment above incorrect?

> Also, please add the use_case_id to the ORDER BY key as well (either before or after tags) but not to the PRIMARY KEY. The partition key should be enough to prevent Clickhosue from merging rows of different retentions but we never know.

And can you explain in more depth what `retention is still missing.` means? You want `retention_days` in the ORDER BY so that rows with different timestamps can't be merged (even though they should always be partitioned separately)?